sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -78,9 +78,10 @@ class ResolveSQLOnFile(sparkSession: SparkSession) extends Rule[LogicalPlan] {
         case e: Exception =>
           // the provider is valid, but failed to create a logical plan
           u.failAnalysis(
-            errorClass = "_LEGACY_ERROR_TEMP_2332",
-            messageParameters = Map("msg" -> e.getMessage),
-            cause = e)
+            errorClass = "UNSUPPORTED_DATASOURCE_FOR_DIRECT_QUERY",
+            messageParameters = Map("dataSourceType" -> u.multipartIdentifier.head),
```

Comment:
this is not just assigning an error class name, but changing the error message as well. It's good to put the data source type in the error message, but I don't think we can simply ignore the original error message `e.getMessage`. @itholic can you fix it?

Reply:
Please fix it before 3.5, as it's a regression

Reply:
are you sure? I don't see it passing the original error message.